lib/sshkit/backends/netssh.rb

```diff
@@ -23,10 +22,27 @@ module SSHKit
   module Backend
 
     class Netssh < Abstract
+      def self.assert_valid_transfer_method!(method)
+        return if [nil, :scp, :sftp].include?(method)
```

Comment:
`nil`? 

I suppose it's inconsequential when `transfer_method` defaults to `:scp` along with the behaviors of `#with_transfer`.

Reply:
@JasonPoll `Host#transfer_method` is initially set to `nil` (which means "use the global default"), so I wanted to make sure that `nil` is an allowable value.

Reply:
...oh wait...I think I understand. `nil` is considered a valid value because per-host configuration could be `nil` and global configuration could be what is specifying `:scp` or `:sftp`. 

Even in the case when both host and global configuration are assigned a `transfer_value` of `nil`, the `#with_transfer` method still embodies the previous behavior of transfers being performed via SCP.

Reply:
Maybe a more correct implementation would be to have `assert_valid_transfer_method!` strictly only allow :scp and :sftp, with a workaround in `Host#transfer_method=` to allow `nil`.

```ruby
# netssh.rb

def self.assert_valid_transfer_method!(method)
  return if [:scp, :sftp].include?(method)
  raise ArgumentError, "#{method.inspect} is not a valid transfer method. Supported methods are :scp, :sftp."
end
```

```ruby
# host.rb

def transfer_method=(method)
  Backend::Netssh.assert_valid_transfer_method!(method) unless method.nil?

  @transfer_method = method
end
```

Reply:
Addressed in 9bbcc2c